### ISSUE TITLE
xrefs: fully normalize DecorationsRequest Locations for line/column offset points [phab:D122]

### ISSUE THREAD

**Issue by kythe** [google/kythe#327]
_Created by **schroederc** at 2015-03-04 19:32:43_: 

This commit also changes the format of the `--span` arguments to the Kythe
command-line tool so that simple `\d+-\d+` format is for line numbers.  To use
byte-offsets, use the point format `b\d+` instead.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2015-03-05 17:59:22_:

Fine as is, though I'd be tempted to use a `switch` here.

```
switch {
case p == nil:
  return np
case p.ByteOffset > 0:
  …
case p.LineNumber > 0:
  …
}
return np
```

Entirely up to you, however.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2015-03-05 17:59:22_:

We'll likely eventually want to memoize some of this work, but that certainly need not be now.

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2015-03-05 17:59:22_

**Comment by kythe** [comment]
_**schroederc** closed via commit 36a2436a1d45eab074d71b5c0ecc5b0177739d95 at 2015-03-05 18:04:40_

**Comment by kythe** [comment]
commit 6155120f9810a944bddc6fc4354cdddd6dd5cf4d
